src/transform/src/nonnull_requirements.rs

```diff
@@ -195,14 +195,55 @@ impl NonNullRequirements {
                 expected_group_size: _,
             } => {
                 let mut new_columns = HashSet::new();
-                for column in columns {
-                    // No obvious requirements on aggregate columns.
-                    // A "non-empty" requirement, I guess?
-                    if column < group_key.len() {
-                        group_key[column].non_null_requirements(&mut new_columns);
-                    }
-                    if column == group_key.len() && aggregates.len() == 1 {
-                        aggregates[0].expr.non_null_requirements(&mut new_columns);
+                for column in columns.iter().filter(|c| **c < group_key.len()) {
+                    group_key[*column].non_null_requirements(&mut new_columns);
+                }
+
+                if columns.iter().any(|c| *c >= group_key.len()) {
+                    // We can convert a non-null requirement on an aggregate function into
+                    // a non-null requirement on its parameter only if we are sure that
+                    // won't affect the result of any other aggregation.
+                    // If there is non-null requirement on MAX(c1), for example, it is
+                    // implied that MIN(c1) won't be null either, so we can safely require
+                    // c1 to be non-null as long as there isn't any other aggregation
+                    // depending on any other column or a COUNT(*).
+                    // Note: all columns from the non-preserving side of an outer join could
+                    // be treated as a single column for this analysis, but we don't have
+                    // that information here.
+                    let new_nonnull_columns_per_aggregate = aggregates
```

Comment:
for each aggregate, see which columns would be have nonnull if the aggregation is nonnull